### PR TITLE
ENH: gaussian_kde.logpdf based on logsumexp

### PR DESCRIPTION
**Pull request by evgenyzhurko**
Solution for issue #6869 
Updated function for computing `gaussian_kde.logpdf` based on `scipy.misc.logsumexp`

### CODE REVIEW

File: scipy/stats/kde.py [hunk 1]
```diff
@@ -527,11 +528,36 @@ def pdf(self, x):
...
+
+        result = zeros((m,), dtype=float)
+
+        if m >= self.n:
+            # there are more points than data, so loop over data
+            for i in range(self.n):
+                diff = self.dataset[:, i, newaxis] - points
+                tdiff = dot(self.inv_cov, diff)
+                energy = sum(diff*tdiff,axis=0) / 2.0
+                result = logsumexp(-energy, b=1/self._norm_factor)
```
Comment (pv) [inline scipy/stats/kde.py]: This looks wrong.

Comment (pv) [inline scipy/stats/kde.py]: This looks wrong.

**Review by pv — CHANGES_REQUESTED** [review]
Needs tests, all branches are not covered.
Cf https://codecov.io/gh/scipy/scipy/compare/c992cd6fa92db803770ae7fc477b610de2099877...a3414585549d631eba210d988f7ed21c7884f540

File: scipy/stats/kde.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 # Scipy imports.
 from scipy._lib.six import callable, string_types
 from scipy import linalg, special
+from scipy.misc import logsumexp
```
Comment (ev-br) [inline scipy/stats/kde.py]: Not: from sciipy.special import logsumexp

File: scipy/stats/tests/test_kdeoth.py [hunk 3]
```diff
@@ -235,6 +235,12 @@ def test_pdf_logpdf():
     logpdf2 = gkde.logpdf(xs)
     assert_almost_equal(logpdf, logpdf2, decimal=12)
 
+    # There are more points than data
+    gkde = stats.gaussian_kde(xs)
+    pdf = np.log(gkde.evaluate(xn))
+    pdf2 = gkde.logpdf(xn)
+    assert_almost_equal(logpdf, logpdf2, decimal=12)
```
Comment (ev-br) [inline scipy/stats/tests/test_kdeoth.py]: assert_almost_equal(PDF, pdf2) or rename variables

**Review by ev-br — CHANGES_REQUESTED** [review]

**Review by ev-br — APPROVED** [review]

**Pull request merged**